### PR TITLE
Add Item Component - Ignore should be deleted

### PR DESCRIPTION
**Pull request by shaibv**
Adding AddItem component and unbinding it from ImageViewer.
This is Phase 1, Phase 2 will deal with dimensions/layout.

### CODE REVIEW

File: src/AddItem/AddItem.spec.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    };
+  });
+
+  describe('when default scenario', () => {
+    beforeEach(() => {
+      props = {
+        onAddItem: addItem,
+        tooltipContent: TOOLTIP_CONTENT
+      };
+      driver = createDriver(<AddItem {...props}/>);
```
Comment (lbelinsk) [inline src/AddItem/AddItem.spec.js]: Don't create the driver beforeEach if you are doing it again in the test itself. 2 mounts...

File: src/AddItem/AddItem.spec.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    });
+
+  });
+
+  describe('hide or show add item', () => {
+
+    it('should have an Add item tooltip', () => {
+      driver = createDriver(<AddItem {...props}/>);
+      const wrapper = driver.getElement();
+      const TooltipDriver = tooltipTestkitFactory({wrapper, dataHook: 'add-tooltip'});
```
Comment (lbelinsk) [inline src/AddItem/AddItem.spec.js]: You can add a method to the driver `.getTooltip()`

File: src/AddItem/AddItem.spec.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    it('should not add style attribute when width and height props are not passed', () => {
+      driver = createDriver(<AddItem {...props}/>);
+      expect(driver.getContainerStyles()).toEqual(null);
+    });
+
+  });
+
+  describe('hide or show add item', () => {
+
+    it('should have an Add item tooltip', () => {
```
Comment (lbelinsk) [inline src/AddItem/AddItem.spec.js]: Do we always want to show the tooltip on hover?
If not, we should have a test for it

File: src/AddItem/AddItem.spec.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+  const TOOLTIP_CONTENT = 'BLA BLA';
+  const createDriver = createDriverFactory(addItemDriverFactory);
+  let props, driver;
+  const addItem = jest.fn();
+
+
+  beforeEach(() => {
+    document.body.innerHTML = '';
+    props = {
```
Comment (lbelinsk) [inline src/AddItem/AddItem.spec.js]: It is hard to follow which props object is actually being rendered on each test. Too many shadowing of this variable.

File: testkit/index.js [hunk 5]
```diff
@@ -9,6 +9,9 @@ export const inputAreaTestkitFactory = testkitFactoryCreator(inputAreaDriverFact
 import tooltipDriverFactory from '../src/Tooltip/Tooltip.driver';
 export const tooltipTestkitFactory = testkitFactoryCreator(tooltipDriverFactory);
 
+import addItemDriverFactory from '../src/AddItem/AddItem.driver';
```
Comment (lbelinsk) [inline testkit/index.js]: Expose also from protractor

File: src/AddItem/AddItem.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+          </Tooltip>
+        </div>
+      </div>
+    );
+  }
+}
+
+AddItem.propTypes = {
+  onAddItem: PropTypes.func,
+  width: PropTypes.number,
```
[the remark this answers is not in the document]
Comment (shaibv) [reply src/AddItem/AddItem.js]: the size props will be changed in the next PR

File: src/AddItem/AddItem.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  render() {
+
+    const {
+      onAddItem,
+      width,
+      height,
+      tooltipContent
+    } = this.props;
+
+    const tooltipCommonProps = {
```
[the remark this answers is not in the document]
Comment (shaibv) [reply src/AddItem/AddItem.js]: It's a generic component it must be

File: src/AddItem/AddItem.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      </div>
+    );
+  }
+}
+
+AddItem.propTypes = {
+  onAddItem: PropTypes.func,
+  width: PropTypes.number,
+  height: PropTypes.number,
+  tooltipContent: PropTypes.string.isRequired
```
[the remark this answers is not in the document]
Comment (shaibv) [reply src/AddItem/AddItem.js]: this code is out of date

File: src/AddItem/AddItem.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+import Tooltip from '../Tooltip';
+import Plus2 from '../Icons/dist/components/Plus2';
+import WixComponent from '../BaseComponents/WixComponent';
+
+class AddItem extends WixComponent {
+
+  render() {
+
+    const {
+      onAddItem,
```
[the remark this answers is not in the document]
Comment (shaibv) [reply src/AddItem/AddItem.js]: I think onAddItem is more descriptive and tells what the function is for.
But if you think onClick is more consistent I'm OK with it.

File: src/AddItem/AddItem.spec.js (hunk 3 above)
Comment (shaibv) [reply src/AddItem/AddItem.spec.js]: when we have content
